tests/src/CLRTest.Execute.Batch.targets

```diff
@@ -213,21 +217,83 @@ IF NOT "%CLRCustomTestLauncher%"=="" (
 )
       ]]></BatchCLRTestLaunchCmds>
       <BatchCLRTestLaunchCmds Condition=" '$(IlasmRoundTrip)'=='true' "><![CDATA[
-$(BatchCLRTestLaunchCmds)
-ECHO %LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
-%LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
-
-IF NOT "!ERRORLEVEL!"=="%CLRTestExpectedExitCode%" (
-  ECHO END EXECUTION OF IL{D}ASM BINARY - FAILED !ERRORLEVEL! vs %CLRTestExpectedExitCode%
-  ECHO FAILED
-  Exit /b 1
+  $(BatchCLRTestLaunchCmds)
+IF NOT DEFINED DoLink ( 
+  ECHO %LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
+  %LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
+
+  IF NOT "!ERRORLEVEL!"=="%CLRTestExpectedExitCode%" (
+    ECHO END EXECUTION OF IL{D}ASM BINARY - FAILED !ERRORLEVEL! vs %CLRTestExpectedExitCode%
+    ECHO FAILED
+    Exit /b 1
+  )
 )
       ]]></BatchCLRTestLaunchCmds>
       <BatchCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun'"><![CDATA[
+
+set ExePath=$(InputAssemblyName)
+set LinkBin=__Link
+set Assemblies=-a System.Private.CoreLib
+set ReflectionRoots=
+
+IF defined DoLink ( 
+    IF NOT EXIST !ILLINK! (
```

Comment:
Would it be possible to move the linker-specific logic to its own section, something like IllinkBatchScript, instead of putting it in BatchCLRTestLaunchCmds? You could have BatchCLRTestLaunchCmds check ExePath and set it to $(InputAssemblyName) if none of the previous steps have modified the ExePath.